cms/djangoapps/modulestore_migrator/models.py

```diff
@@ -0,0 +1,223 @@
+from django.utils.translation import gettext_lazy as _
+from user_tasks.models import UserTaskStatus
+
+from model_utils.models import TimeStampedModel
+from opaque_keys.edx.django.models import (
+    LearningContextKeyField,
+    UsageKeyField,
+)
+from openedx_learning.api.authoring_models import (
+    LearningPackage, PublishableEntity, Collection, DraftChangeLog, DraftChangeLogRecord
+)
+
+from .data import CompositionLevel, RepeatHandlingStrategy
+
+User = get_user_model()
+
+
+class ModulestoreSource(models.Model):
+    """
+    A legacy learning context (course or library) which can be a source of a migration.
+    """
+    key = LearningContextKeyField(
+        max_length=255,
+        unique=True,
+        help_text=_('Key of the content source (a course or a legacy library)'),
+    )
+    forwarded = models.OneToOneField(
+        'modulestore_migrator.ModulestoreMigration',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        help_text=_('If set, the system will forward references of this source over to the target of this migration'),
+        related_name="forwards",
+    )
+
+    def __str__(self):
+        return f"{self.__class__.__name__}('{self.key}')"
+
+    __repr__ = __str__
+
+
+class ModulestoreMigration(models.Model):
+    """
+    Tracks the action of a user importing a Modulestore-based course or legacy library into a
+    learning-core based learning package
+
+    Notes:
+    * As of Ulmo, a learning package is always associated with a v2 content library, but we
+      will not bake that assumption into this model)
+    * Each Migration is tied to a single UserTaskStatus, which connects it to a user and
+      contains the progress of the import.
+    * A single ModulestoreSource may very well have multiple ModulestoreMigrations; however,
+      at most one of them with be the "authoritative" migration, as indicated by `forwarded`.
+    """
+
+    ## MIGRATION SPECIFICATION
+    source = models.ForeignKey(
+        ModulestoreSource,
+        on_delete=models.CASCADE,
+        related_name="migrations",
+    )
+    source_version = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        help_text=_('Migrated content version, the hash of published content version'),
+    )
+    composition_level = models.CharField(
+        max_length=255,
+        choices=CompositionLevel.supported_choices(),
+        default=CompositionLevel.Component.value,
+        help_text=_('Maximum hierachy level at which content should be aggregated in target library'),
+    )
+    repeat_handling_strategy = models.CharField(
+        choices=RepeatHandlingStrategy.supported_choices(),
+        default=RepeatHandlingStrategy.default().value,
+        max_length=24,
+        help_text=_(
+            "If a piece of content already exists in the content library, choose how to handle it."
+        ),
+    )
+    preserve_url_slugs = models.BooleanField(
+        default=True,
```

Comment:
```suggestion
        default=False,
```
Let's default to False, since existing URL slugs are mostly bad :)